internal/credential/vault/repository_credentials.go

```diff
@@ -75,18 +70,28 @@ func (r *Repository) IssueCredentials(ctx context.Context, sessionId string, lib
 		cred.PublicId = credId
 		cred.IsRenewable = secret.Renewable
 
-		query, queryValues := cred.insertQuery()
+		insertQuery, insertQueryValues := cred.insertQuery()
+		updateQuery, updateQueryValues := cred.updateSessionQuery(lib.Purpose)
 		if _, err := r.writer.DoTx(ctx, db.StdRetryCnt, db.ExpBackoff{},
 			func(_ db.Reader, w db.Writer) error {
-				rowsInserted, err := r.writer.Exec(ctx, query, queryValues)
+				rowsInserted, err := w.Exec(ctx, insertQuery, insertQueryValues)
 				switch {
 				case err == nil && rowsInserted > 1:
 					return errors.New(errors.MultipleRecords, op, "more than 1 credential would have been inserted")
 				case err != nil:
 					return errors.Wrap(err, op)
-				default:
-					return nil
 				}
+
+				rowsUpdated, err := w.Exec(ctx, updateQuery, updateQueryValues)
+				switch {
+				case err == nil && rowsUpdated == 0:
+					return errors.New(errors.InvalidDynamicCredential, op, "no matching dynamic credential for session found")
+				case err == nil && rowsUpdated > 1:
+					return errors.New(errors.MultipleRecords, op, "more than 1 session credential would have been updated")
+				case err != nil:
```

Comment:
Why not check `err != nil` first and then drop the `err == nil &&` from the other cases?

Reply:
Good idea. Updated. Ty!